app/forms/manage_users_form.rb

```diff
@@ -14,28 +14,34 @@ def initialize(team)
   def submit(user_emails:)
     # Devise makes emails case insensitive by downcasing prior to saving
     @user_emails = user_emails.map(&:downcase)
+    @emails_to_remove = team.users.pluck(:email) - @user_emails
+    @emails_to_add = @user_emails - team.users.pluck(:email)
 
     return false unless valid?
 
-    result = team.update(users: users_from_user_emails)
-    errors.merge!(team.errors)
-    result
+    ActiveRecord::Base.transaction do
+      destroy_removed_user_teams
+      add_new_users
+    end
+    true
+  rescue ActiveRecord::ActiveRecordError => e
+    errors.add(:base, "Unable to update users: #{e.message}") 
+    false
   end
 
   private
 
-  def users_from_user_emails
-    missing_users = missing_emails.map { |e| User.new(email: e) }
-    (existing_users + missing_users).sort_by(&:email)
-  end
+  attr_reader :emails_to_remove, :emails_to_add
 
-  def missing_emails
-    existing_emails = existing_users.map(&:email)
-    user_emails.filter { |email| !existing_emails.include? email }
+  def destroy_removed_user_teams
+    users_to_remove = User.where(email: emails_to_remove).pluck(:id)
+    UserTeam.where(group_id: team.id, user_id: users_to_remove).each do |ut|
+      ut.destroy! # we want to trigger the paper_trail callback
+    end
```

Comment:
does the `has_many :through` do this? Is there any way we can get this desired behavior through `dependent: :destroy` annotations?

Reply:
I tried - it didn't work unfortunately. The records are actually being destroyed but PaperTrail wasn't being triggered.

I'm 99% sure I tested this with the feature spec before I started messing with the Form Object but I'm going to check one more time.

Reply:
@zachmargolis turns out I forgot to test this again after I remembered to manually enable PaperTrail for the feature spec - that indeed works when applied to the `has_many :users` association. Thanks so much!